python_modules/dagster/dagster/_core/pipes/subprocess.py

```diff
@@ -108,11 +111,19 @@ def run(
                     **pipes_session.get_bootstrap_env_vars(),
                 },
             )
-            process.wait()
-            if process.returncode != 0:
-                raise DagsterPipesExecutionError(
-                    f"External execution process failed with code {process.returncode}"
-                )
+            try:
+                process.wait()
+                if process.returncode != 0:
+                    raise DagsterPipesExecutionError(
+                        f"External execution process failed with code {process.returncode}"
+                    )
+            except DagsterExecutionInterruptedError:
+                context.log.info("[pipes] execution interrupted, sending SIGINT to subprocess.")
+                # send sigint to give external process chance to exit gracefully
+                process.send_signal(signal.SIGINT)
```

Comment:
the alternative choice of using `process.terminate()` which uses `SIGTERM` would be having `open_dagster_pipes` install a `SIGTERM` handler since the default no-handler behavior is immediate process halt